### PR TITLE
feat: display product reviews & refactor reviews and ratings into an extension

### PR DESCRIPTION
**Pull request by SGrueber**
## PR Type

[x] Feature
[x] Refactoring (no functional changes, no API changes)

## What Is the Current Behavior?
On product detail page product reviews are not display.
The number behind the rating stars of a product displays the average rating (bug).

## What Is the New Behavior?
On product detail page the user can click a reviews tab to see the review comments of other users. Creating a review comment is still pending and will be done soon.
The number behind the rating stars displays the number of ratings.
The rating feature is implemented in an extension.

## Does this PR Introduce a Breaking Change?

[x] Yes

### CODE REVIEW

File: src/assets/i18n/fr_FR.json [hunk 1]
```diff
@@ -884,6 +884,9 @@
   "product.ready_for_shipment.within24": "Habituellement prêt à être expédié dans les 24 heures",
   "product.remove_all_product_filters.text": "Réinitialiser tous les filtres",
   "product.retailSet.products.heading": "Ce jeu de produits contient",
+  "product.reviews.heading": "Notations",
```
Comment (NicolasGrizard) [inline src/assets/i18n/fr_FR.json:887]: *Avis

File: src/assets/i18n/fr_FR.json [hunk 2]
```diff
@@ -884,6 +884,9 @@
...
+  "product.reviews.no_reviews_yet.text": "Aucune critique n'a encore été soumise.",
```
Comment (NicolasGrizard) [inline src/assets/i18n/fr_FR.json:888]: *Aucun avis n'a encore été soumis.

**Review by NicolasGrizard — CHANGES_REQUESTED** [review]

File: src/app/core/facades/product-context.facade.ts [hunk 3]
```diff
@@ -113,6 +114,8 @@ export interface ProductContext {
   // child contexts
   propagateActive: boolean;
   children: Record<string | number, ProductContext>;
+
+  reviews: ProductReview[];
```
Comment (dhhyi) [inline src/app/core/facades/product-context.facade.ts:118]: Should not be part of this facade

File: src/app/core/models/product-reviews/product-review.model.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export interface ProductReview {
```
Comment (dhhyi) [inline src/app/core/models/product-reviews/product-review.model.ts:1]: Should be moved to the extension

**Review by dhhyi — CHANGES_REQUESTED** [review]

**Comment by SGrueber** [comment]
There is still an issue left: If the user opens the reviews tab the reviews are fetched from the server twice. This has to be solved before the PR can be merged.

File: src/app/extensions/rating/shared/product-rating-star/product-rating-star.component.ts [hunk 5]
```diff
@@ -12,6 +14,7 @@ import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
   styleUrls: ['./product-rating-star.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
+@GenerateLazyComponent()
```
Comment (dhhyi) [inline src/app/extensions/rating/shared/product-rating-star/product-rating-star.component.ts:17]: Not needed, right? 🤔

File: src/app/extensions/rating/rating.module.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import { SharedModule } from 'ish-shared/shared.module';
+
+import { ProductRatingStarComponent } from './shared/product-rating-star/product-rating-star.component';
+import { ProductRatingComponent } from './shared/product-rating/product-rating.component';
+import { ProductReviewsComponent } from './shared/product-reviews/product-reviews.component';
+
+@NgModule({
+  imports: [SharedModule],
+  declarations: [ProductRatingComponent, ProductRatingStarComponent, ProductReviewsComponent],
+  exports: [ProductRatingComponent, ProductRatingStarComponent, ProductReviewsComponent, SharedModule],
```
Comment (dhhyi) [inline src/app/extensions/rating/rating.module.ts:12]: `ProductRatingStarComponent` doesn't need export, I think

**Review by dhhyi — CHANGES_REQUESTED** [review]

**Review by shauke — DISMISSED** [review]

**Comment by shauke** [comment]
> There is still an issue left: If the user opens the reviews tab the reviews are fetched from the server twice. This has to be solved before the PR can be merged.

Issue was fixed by removing doubled `productReviews$ | async ` pipe.

**Review by dhhyi — APPROVED** [review]
:+1:

**Pull request merged**